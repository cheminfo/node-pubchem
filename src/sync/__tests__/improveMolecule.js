'use strict';

const fs = require('fs-extra');

const improveMolecule = require('../improveMolecule');

test('creation of molecular formula', () => {
  let molfile = fs.readFileSync(`${__dirname}/test.mol`, 'utf8');
  let molecule = {
    PUBCHEM_COMPOUND_CID: 1234,
    molfile
  };
  let result = improveMolecule(molecule);
  expect(result).toMatchSnapshot();
});

