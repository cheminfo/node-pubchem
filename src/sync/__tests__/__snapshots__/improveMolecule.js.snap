// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creation of molecular formula 1`] = `
Object {
  "_id": 1234,
  "atom": Object {
    "C": 4,
    "H": 11,
  },
  "charge": -1,
  "em": 60.0894301897,
  "iupac": undefined,
  "mf": "C2H5(-).CH6[13C]",
  "mw": 60.122911,
  "nbFragments": 2,
  "ocl": Object {
    "coordinates": "!B_qL]pW|S",
    "id": "gB\`AMrQt@qAX@",
    "index": Array [
      1073741824,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
  },
  "seq": 0,
  "unsaturation": 0,
}
`;
